lib/bitswap.go

```diff
@@ -21,7 +22,27 @@ type BsCheckOutput struct {
 	Error     error
 }
 
-func CheckBitswapCID(ctx context.Context, c cid.Cid, ma multiaddr.Multiaddr) (*BsCheckOutput, error) {
+func (o *BsCheckOutput) MarshalJSON() ([]byte, error) {
+	var errorMsg *string
+	if o.Error != nil {
+		m := o.Error.Error()
+		errorMsg = &m
+	}
+	anon := struct {
+		Found     bool
+		Responded bool
+		Error     *string
```

Comment:
We could just switch BsCheckOutput to have Error be type String instead of doing this dance (which probably Go should just do for us, but 🤷 ). Wasn't sure if this would break anything @whyrusleeping was using though.